src/hotspot/share/runtime/objectMonitor.hpp

```diff
@@ -302,6 +302,22 @@ class ObjectMonitor : public CHeapObj<mtInternal> {
   // returns false and throws IllegalMonitorStateException (IMSE).
   bool      check_owner(TRAPS);
 
+ private:
+  class ExitOnSuspend {
+   protected:
+    ObjectMonitor* _om;
+    bool _om_op_done;
+   public:
+    ExitOnSuspend(ObjectMonitor* om) : _om(om), _om_op_done(false) {}
+    void operator()(JavaThread* current);
+    bool om_op_done() { return _om_op_done; }
+  };
+  class ClearSuccOnSuspend : public ExitOnSuspend {
```

Comment:
I don't see why there is any relationship between these two. You don't clear-succ and exit.

Reply:
Fixed